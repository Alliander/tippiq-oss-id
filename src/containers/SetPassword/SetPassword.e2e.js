export const page = {
  isLoaded: () => protractor.ExpectedConditions.presenceOf(element(by.id('page-set-password'))),
  heading: () => element(by.css('h1')).getText(),
  submit: () => element(by.id('passwordSubmitButton')).click(),
  passwordField: () => element(by.id('password')),
  passwordError: () => element(by.id('password'))
    .element(by.xpath('following-sibling::div')).getText(),
  error: () => element(by.css('.alert-danger')),
  result: () => element(by.id('successMessage')),
};

describe('Set password container', () => {
  it('should set a new password', () => {
    browser.get('/wachtwoord-instellen');
    browser.wait(page.isLoaded(), 10000);

    expect(page.heading()).toEqual('Wachtwoord kiezen');

    // Check required field
    page.submit();
    expect(page.passwordError()).toEqual('Verplicht');

    // Check submit
    page.passwordField().sendKeys('test1234ABC');
    page.submit();

    // Check that submitting without a token fails
    browser.wait(protractor.ExpectedConditions.presenceOf(page.error()), 10000);
    expect(page.error().getText()).toEqual('De verificatiecode is verlopen');
  });
});
