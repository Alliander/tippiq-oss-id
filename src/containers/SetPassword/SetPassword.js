/**
 * Set password container.
 * @module components/SetPassword
 */

import React, { Component, PropTypes } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { get } from 'lodash';

import { ChangePasswordForm, PageTitle } from '../../components';
import { setPassword } from '../../actions';

import styles from './SetPassword.scss';


/**
 * Set password container class.
 * @class SetPassword
 * @extends Component
 */
export class SetPassword extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    setPassword: PropTypes.func.isRequired,
    location: PropTypes.object,
    password: PropTypes.object.isRequired,
  }

  /**
   * Component will mount
   * @method componentWillMount
   * @returns {undefined}
   */
  componentWillMount() {
    const token = get(this.props, 'location.query.token');
    if (token) {
      this.setState({ token });
    }
  }

  /**
   * Handler for submit action.
   * @method handleSubmit
   * @param {Object} data Form data.
   * @returns {undefined}
   */
  handleSubmit = (data) => {
    this.props.setPassword(data.password, get(this.state, 'token'));
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { password } = this.props;
    const errorMessage = password.error ? password.error.message : null;

    const successView = password.save && password.save.success ? (
      <div>
        <div className="row">
          <PageTitle>
            <div className={`col-xs-1 ${styles.icon}`}>
              <i className="fa fa-save fa-3x" />
            </div>
            <div className={`col-xs-6 ${styles.title}`}>
              <span id="successMessage" className="text-primary">Gelukt!</span>
              <br />Wachtwoord opgeslagen.
            </div>
          </PageTitle>
        </div>
        <div className="row">
          <div className="col-xs-12">
            <br />
            <p>
              Vanaf nu kan je inloggen met je nieuwe wachtwoord om je huisregels te beheren.
            </p>
          </div>
        </div>
      </div>) : null;

    return (
      <div id="page-set-password">
        <Helmet title="Wachtwoord kiezen" />
        { successView ||
          <div>
            <div className="row">
              <div className="col-xs-12">
                <PageTitle>
                  <span className="text-primary">Wachtwoord</span> kiezen
                </PageTitle>
              </div>
            </div>
            <div className="row">
              <div className="col-xs-12">
                <p>
                  Kies een wachtwoord voor je bewonersaccount.
                </p>
                <br />
              </div>
            </div>
            <div className="row">
              <ChangePasswordForm
                onSubmit={this.handleSubmit}
                errorMessage={errorMessage}
              />
            </div>
          </div>
        }
      </div>
    );
  }
}

export default connect(
  state => ({
    password: state.password,
  }), dispatch => bindActionCreators({ setPassword }, dispatch)
)(SetPassword);
