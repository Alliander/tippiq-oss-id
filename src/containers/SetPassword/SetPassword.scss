.icon {
  min-width: 110px;
}

.title {
  padding-top: 10px;
  line-height: 1.3em;
}
