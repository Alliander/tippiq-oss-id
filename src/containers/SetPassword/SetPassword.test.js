import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { SetPassword } from './SetPassword';

test('Basic render snapshot test', () => {
  const component = shallow(<SetPassword
    error={'error'}
    setPassword={() => {}}
    location={{ query: '' }}
    password={{
      save: {
        success: '',
      },
      error: {
        message: '',
      },
    }}
  />);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

