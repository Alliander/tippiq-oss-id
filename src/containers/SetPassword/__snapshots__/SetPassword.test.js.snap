// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<div
  id="page-set-password"
>
  <HelmetWrapper
    title="Wachtwoord kiezen"
  />
  <div>
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <PageTitle>
          <span
            className="text-primary"
          >
            Wachtwoord
          </span>
           kiezen
        </PageTitle>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <p>
          Kies een wachtwoord voor je bewonersaccount.
        </p>
        <br />
      </div>
    </div>
    <div
      className="row"
    >
      <ConnectedForm
        errorMessage=""
        onSubmit={[Function]}
      />
    </div>
  </div>
</div>
`;
